### DISCUSSION TITLE
Discussion: Ability to remove Auth requirements from login

### DISCUSSION THREAD

**Discussion by SAS-1** [alexjustesen/speedtest-tracker#391] [Ideas]
Hi All

I have just moved over from https://github.com/henrywhitaker3/Speedtest-Tracker and I see that at the moment, and potentially ongoing, we need to login using email/password. 

I can't speak for everyone and I read that future plans mean this is staying as is, but I would like to have a discussion into why its needed if possible?

In my use case I don't want to login, I am a single user, it's on my LAN and I have no intention of exposing it to the net via reverse proxy or directly so having a requirement for this is annoying.

In my setup if I do use auth it's username/password which the password manager handles then 2FA and IP whitelists so its adding additional complexity for me - plus upcoming changes to chromium browsers  mean unsecured content give an error when filling in details

![image](https://user-images.githubusercontent.com/47122073/216811934-b4e2aabf-ac5a-46b5-ab9e-d04d9386fe4f.png)

I understand if there is a specific reason for keeping this as a requirement but for me I would rather have an ability to turn it off.

Thanks

**Comment by alexjustesen** [comment]
Currently I removed this from the roadmap as removing auth would take a bit of work. Not saying it won't happen because I too don't _need_ authentication it's just not a priority at the moment.

**Comment by Bergruebe** [comment]
Maybe an easy step in this direction is to optional allow the `Queue Speedtest` button for all visitors without login.

**Comment by alexjustesen** [comment]
"Public" dashboards are now available and "Public" results are planned. IMO any function that results in an action should stay behind auth so going to close this one out.

**Comment by Waldorf3** [comment]
Agree with this, all.   Would really like a default page showing the status without any options to change or do anything.  Any updates should of course be authenticated, but the content of the dashboard graphs with speed etc really doesn't need to be secret. At least make it optional, annoying to have to login every time I just want to check if the internet works ok.
